Sources/OAApplicationMode.m

```diff
@@ -601,12 +601,48 @@ + (void) saveProfile:(OAApplicationMode *)appMode
     else if (![_values containsObject:appMode])
     {
         [_values addObject:appMode];
+        [OAApplicationMode saveBackupOfAppMode: appMode];
     }
     
     [self reorderAppModes];
     [self saveCustomAppModesToSettings];
 }
 
++ (void) saveBackupOfAppMode:(OAApplicationMode *)appMode
+{
+    NSFileManager *fileManager = NSFileManager.defaultManager;
+    NSString *backupFolderPath = [[NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES) firstObject] stringByAppendingPathComponent:@"osfBackup"];
+    
+    BOOL isDir = YES;
+    if (![fileManager fileExistsAtPath:backupFolderPath isDirectory:&isDir])
+          [fileManager createDirectoryAtPath:backupFolderPath withIntermediateDirectories:YES attributes:nil error:nil];
+    
+    if (appMode.stringKey)
+    {
+        NSString *backupFilePath = [[backupFolderPath stringByAppendingPathComponent:appMode.stringKey] stringByAppendingPathExtension:@"plst"];
+        [appMode.toJson writeToFile:backupFilePath atomically:YES];
```

Comment:
Not only appMode.toJson should be saved, but all profile depended settings! Use OASettingsExporter.